init_process.go

```diff
@@ -0,0 +1,409 @@
+	mainConfigFile, err := os.Create(filePath)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalln("Cannot create main config file")
+	}
+	defer mainConfigFile.Close()
+}
+
+//add @INCLUDE original main config file to the new main config file
+func includeOriginalMainConfigFile(mainConfigFilePath, originalMainConfigFilePath string) {
+	mainConfigFile, err := os.OpenFile(mainConfigFilePath, os.O_APPEND|os.O_WRONLY, 0777)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Unable to read main config file: %s\n", mainConfigFilePath)
+	}
+	defer mainConfigFile.Close()
+
+	writeContent := "@INCLUDE " + originalMainConfigFilePath + "\n"
+	_, err = mainConfigFile.WriteString(writeContent)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Cannot write %s in main config file: %s\n", writeContent[:len(writeContent)-2], mainConfigFilePath)
+	}
+}
+
+//create fluent bit command to use new main config file
+func createCommand(command *string, filePath string) {
+	*command = *command + " -c /" + filePath
+}
+
+//create a folder to store config files user specified
+func createConfigFileFolder(folderPath string) {
+	os.Mkdir(folderPath, os.ModePerm)
+}
+
+//get config files user specified according to environment variables
+func getAllConfigFiles() {
+
+	//get all env vars
+	envs := os.Environ()
+	//find all env vars match specified prefix
+	for _, env := range envs {
+		var envKey string
+		var envValue string
+		env_kv := strings.SplitN(env, "=", 2)
+		if len(env_kv) != 2 {
+			log.Printf("Unrecognizable environment variables: %s\n", env)
+			continue
+		}
+
+		envKey = string(env_kv[0])
+		envValue = string(env_kv[1])
+
+		matched_s3, _ := regexp.MatchString("aws_fluent_bit_s3_", envKey)
+		matched_file, _ := regexp.MatchString("aws_fluent_bit_file_", envKey)
+		//if this env var's value is an arn
+		if matched_s3 {
+			getS3ConfigFile(envValue)
+		}
+		//if this env var's value is a path of our built-in config file
+		if matched_file {
+			getBuiltInConfigFile(envValue)
+		}
+	}
+}
+
+//copy built-in config file to config file folder
+func getBuiltInConfigFile(path string) {
+
+	fileName := strings.SplitN(path, "/", -1)
+
+	src, err := os.Open(path[1:]) //remove first "/"
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Cannot find built-in config file: %s\n", path[1:])
+	}
+	defer src.Close()
+
+	dstPath := ConfigFileFolderPath + "/" + fileName[len(fileName)-1]
+	dst, err := os.Create(dstPath)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalln("Cannot creat config file folder")
+	}
+	defer dst.Close()
+
+	io.Copy(dst, src)
+}
+
+//download S3 config file to config file folder
+func getS3ConfigFile(arn string) {
+	if !existS3er {
+		// creat global s3 client
+		// s3er = s3.New(session.Must(session.NewSession()))
+		s3er = s3.New(session.Must(session.NewSession(&aws.Config{
+			//if not specify region here, missingregion error will raise when get bucket location
+			//can be any region
+			Region: aws.String("us-east-2"),
```

Comment:
I think it would be better if the region is the same as `metadata.region`? like we don't specify it. Or it can be if we have ecs metadata, then it is `metadata.region`, Otherwise it can be us-east-1

Reply:
Yeah make sense, will modify